src/wait.ts

```diff
@@ -34,11 +40,18 @@ export function waitFor<T>(getter: () => T, options: {
     return new Promise(rs => setTimeout(rs, options.interval)).then(wait);
   }
 
+  let stack: any;
+  try {
+    throw new Error();
+  } catch (e) {
+    stack = e.stack;
```

Comment:
The trick here is capturing the stack before we do some hectic async anonymous function stuff.